src/algorithms/types/Param.types.ts

```diff
@@ -61,6 +61,21 @@ export interface AllParams {
 
 export type AllParamsFlat = PopulationData & EpidemiologicalData & SimulationData & ContainmentData
 
+// part of the application state to be persisted in the URL
+export interface PersistedState {
+  current: string
+  data: ScenarioData
+}
+
+// format of the application state persisted in the URL
+export interface PersistedStateDto {
+  current: string
+  population: PopulationData
+  epidemiological: EpidemiologicalData
+  simulation: SimulationData
+  containment: TimeSeries
+}
```

Comment:
Since you are adding this I think it would be a good point to address the fact that `current` is irrelevant. What do you think we should do?

Reply:
It might be irrelevant if the user custoized some params. But in general, it isn't.

I think `curren`t should still be send over the wire (via URL). Because, if i want to select one of the existing, default scenarios (say "Croatia") and I want to send the link to someone, they should see "Croatia" selected in the dropdown.

Basically, we need to persist all **the state needed to reconstruct the current screen**. Without persisting `current` you can't do that.

Reply:
Hmm it might not be irrelevant to the user who is sharing their parameters (ie. when they don't modify any params like you said) but the value is irrelevant upon deserialization. It's true that removing `current` will make it so we can't reconstruct an identical screen but doing so is not technically correct (the state is being initialized from the URL parameters, not the scenario defaults).

Reply:
Then we should really consider not to have the scenario as a dropdown.

We should really have a "Load existing scenario" button, which opens a popup window with a number of scenarios to load: https://github.com/neherlab/covid19_scenarios/issues/397